euphonic/writers/phonon_website.py

```diff
@@ -0,0 +1,218 @@
+    https://henriquemiranda.github.io/phononwebsite
+
+    Parameters
+    ----------
+    output_file
+        Path to output file
+    name
+        Set "name" metadata, to be used as figure title
+    x_tick_labels
+        index and label for high symmetry labels (if known)
+
+    """
+
+    with open(output_file, 'w') as fd:
+        json.dump(_modes_to_phonon_website_dict(modes=modes,
+                                                name=name,
+                                                x_tick_labels=x_tick_labels),
+                  fd)
+
+
+def _crystal_website_data(crystal: Crystal) -> dict[str, Any]:
+    elements = [
+        '_', 'H', 'He', 'Li', 'Be', 'B', 'C', 'N', 'O', 'F', 'Ne', 'Na',
+        'Mg', 'Al', 'Si', 'P', 'S', 'Cl', 'Ar', 'K', 'Ca', 'Sc', 'Ti', 'V',
+        'Cr', 'Mn', 'Fe', 'Co', 'Ni', 'Cu', 'Zn', 'Ga', 'Ge', 'As', 'Se',
+        'Br', 'Kr', 'Rb', 'Sr', 'Y', 'Zr', 'Nb', 'Mo', 'Tc', 'Ru', 'Rh',
+        'Pd', 'Ag', 'Cd', 'In', 'Sn', 'Sb', 'Te', 'I', 'Xe', 'Cs', 'Ba',
+        'La', 'Ce', 'Pr', 'Nd', 'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho',
+        'Er', 'Tm', 'Yb', 'Lu', 'Hf', 'Ta', 'W', 'Re', 'Os', 'Ir', 'Pt',
+        'Au', 'Hg', 'Tl', 'Pb', 'Bi', 'Po', 'At', 'Rn', 'Fr', 'Ra', 'Ac',
+        'Th', 'Pa', 'U', 'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm',
+        'Md', 'No', 'Lr', 'Rf', 'Db', 'Sg', 'Bh', 'Hs', 'Mt', 'Ds', 'Rg',
+        'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og']
+
+    def get_z(symbol: str) -> int:
+        try:
+            return elements.index(symbol)
+        except ValueError:  # Symbol not found
+            return 0
+
+    def symbols_to_formula(symbols: list[str]) -> str:
+        symbol_counts = Counter(symbols)
+
+        return "".join(f"{symbol}{symbol_counts[symbol]}"
+                       for symbol in sorted(symbol_counts))
+
+    return dict(
+        natoms=len(crystal.atom_type),
+        lattice=crystal.cell_vectors.to("angstrom").magnitude.tolist(),
+        atom_types=crystal.atom_type.tolist(),
+        atom_numbers=list(map(get_z, crystal.atom_type)),
+        formula=symbols_to_formula(crystal.atom_type),
+        atom_pos_red=crystal.atom_r.tolist(),
+        atom_pos_car=(crystal.atom_r @ crystal.cell_vectors
+                      ).to("angstrom").magnitude.tolist()
+    )
+
+
+def _remove_breaks(distances: np.ndarray, btol: float = 10.) -> list[int]:
+    """Collapse large breaks in cumulative-distance array
+
+    These correspond to discontinuous regions of the x-axis: in euphonic
+    plotting this is usually handled by splitting the spectrum and plotting
+    to new axes, but Phonon Website does not handle this.
+
+    Data is modified in-place
+
+    A list of identified breakpoints is returned
+
+    """
+    diff = np.diff(distances)
+    median = np.median(diff)
+    breakpoints = np.where((diff / median) > btol)[0] + 1
+
+    for breakpoint in reversed(breakpoints):
+        distances[breakpoint:] -= (distances[breakpoint]
```

Comment:
Not using `diff`? It this because it's the backwards diff?

Reply:
```python
    for breakpoint in reversed(breakpoints):
        distances[breakpoint:] -= diff[breakpoint - 1]
```
seems to work correctly